docker/pravega/scripts/init_kubernetes.sh

```diff
@@ -0,0 +1,76 @@
+    local cacert="/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+    if [ -z "$namespace" ]; then
+        retval=$( curl --cacert ${cacert} -H "Authorization: Bearer ${bearer}" https://kubernetes/api/v1/${resource_type}/${resource_name} 2> /dev/null | jq -rM "${jsonpath}" 2> /dev/null )
+    else
+        retval=$( curl --cacert ${cacert} -H "Authorization: Bearer ${bearer}" https://kubernetes/api/v1/namespaces/${namespace}/${resource_type}/${resource_name} 2> /dev/null | jq -rM "${jsonpath}" 2> /dev/null )
+    fi
+
+    echo "$retval"
+}
+
+init_kubernetes() {
+    if [ -z "${POD_NAMESPACE}" ]; then
+      echo "POD_NAMESPACE variable not set. Exiting..."
+      exit 1
+    fi
+
+    if [ -z "${POD_NAME}" ]; then
+      echo "POD_NAME variable not set. Exiting..."
+      exit 1
+    fi
+
+    local ns=${POD_NAMESPACE}
+    local podname=${POD_NAME}
+
+    if [ -d "/var/run/secrets/kubernetes.io" ] && [ ! -z "${K8_EXTERNAL_ACCESS}" ]; then
+        echo "Running in a Kubernetes environment and managed by the Pravega Operator with external access enabled"
+        echo "Trying to obtain the external service endpoint..."
+
+        service_type=$( k8 "${ns}" "services" "${podname}" ".spec.type" )
+        if [ "${service_type}" == "LoadBalancer" ]; then
+            export PUBLISHED_ADDRESS=$( k8 "${ns}" "services" "${podname}" ".status.loadBalancer.ingress[].ip" )
+            export PUBLISHED_PORT=$( k8 "${ns}" "services" "${podname}" ".spec.ports[].port" )
+            while [ -z ${PUBLISHED_ADDRESS} ] || [ -z ${PUBLISHED_PORT} ]
+            do
+                echo "LoadBalancer external address and port could not be obtained. Waiting 30 seconds and trying again..."
+                sleep 30
+                export PUBLISHED_ADDRESS=$( k8 "${ns}" "services" "${podname}" ".status.loadBalancer.ingress[].ip" )
+                export PUBLISHED_PORT=$( k8 "${ns}" "services" "${podname}" ".spec.ports[].port" )
+            done
+        elif [ "${service_type}" == "NodePort" ]; then
+            nodename=$( k8 "${ns}" "pods" "${podname}" ".spec.nodeName" )
+            export PUBLISHED_ADDRESS=$( k8 "" "nodes" "${nodename}" ".status.addresses[] | select(.type == \"ExternalIP\") | .address" )
+            export PUBLISHED_PORT=$( k8 "${ns}" "services" "${podname}" ".spec.ports[].nodePort" )
+            while [ -z ${PUBLISHED_ADDRESS} ] || [ -z ${PUBLISHED_PORT} ]
+            do
+                echo "NodePort external address and port could not be obtained. Waiting 30 seconds and trying again..."
+                sleep 30
+                nodename=$( k8 "${ns}" "pods" "${podname}" ".spec.nodeName" )
+                export PUBLISHED_ADDRESS=$( k8 "${ns}" "" "${nodename}" '.status.addresses[] | select(.type == "ExternalIP
+    ") | .address" )' )
+                export PUBLISHED_PORT=$( k8 "${ns}" "services" "${podname}" ".spec.ports[].nodePort" )
+            done
+        else
```

Comment:
Observing that there could be a types could be the following
ExternalName, ClusterIP, NodePort, and LoadBalancer.
ref: https://github.com/kubernetes-client/java/blob/e43a7d160e12ded0eeed033e5b43d32fb8c8d368/kubernetes/docs/V1ServiceSpec.md
Should we handle them too?
ref: https://github.com/kubernetes-client/java/blob/e43a7d160e12ded0eeed033e5b43d32fb8c8d368/kubernetes/docs/V1ServiceSpec.md

Reply:
At this moment we don't support external access with service types other than `LoadBalancer` and `NodePort`. `ClusterIP` is the default service type and we use when external access is disabled. The rest of service types require extra configuration and are not supported in PKS nor there's a requirement to support them.